pages/api/mech-metrics.js

```diff
@@ -91,6 +95,76 @@ const fetchMechRequestsFromSubgraphs = async () => {
   }
 };
 
+// Classified totals derived from subgraphs (Mech + Mech-Marketplace Gnosis + Base)
+const fetchCategorizedRequestTotals = async () => {
+  const predictTraderIds = MECH_AGENT_CLASSIFICATION.predict;
+  const contributeIds = MECH_AGENT_CLASSIFICATION.contribute;
+  const governatooorIds = MECH_AGENT_CLASSIFICATION.governatooor;
+  const allIds = [...predictTraderIds, ...contributeIds, ...governatooorIds];
+
+  try {
+    const [mechResult, marketplaceGnosisResult, marketplaceBaseResult] =
+      await Promise.allSettled([
+        ATA_GRAPH_CLIENTS.legacyMech.request(
+          mechRequestsPerAgentOnchainsQuery(allIds.map(String)),
+        ),
+        ATA_GRAPH_CLIENTS.gnosis.request(
+          mechMarketplaceRequestsPerAgentsQuery(allIds.map(String)),
+        ),
+        ATA_GRAPH_CLIENTS.base.request(
+          mechMarketplaceRequestsPerAgentsQuery(allIds.map(String)),
+        ),
+      ]);
+
+    const combinedCounts = new Map();
+
+    const addCounts = (records) => {
+      if (!Array.isArray(records)) return;
+      records.forEach((item) => {
+        const rawAgentId = item?.id;
+        let agentId;
+        if (typeof rawAgentId === 'number') {
+          agentId = rawAgentId;
+        } else if (typeof rawAgentId === 'string') {
+          const match = rawAgentId.match(/\d+/);
+          agentId = match ? Number(match[0]) : NaN;
+        }
+        if (!Number.isFinite(agentId)) return;
```

Comment:
can you please give me an example of a string agent id that represents not a number?
in both marketplace and legacy mech I just see such ids like `"12"`, or `"5"` etc. so why should we complicate the logic here instead of just simply doing the following
```suggestion
      records.forEach((item) => {
        if (item?.id) return;
        const agentId = Number(item.id);
```